packages/subgraph/src/mapping.ts

```diff
@@ -738,15 +750,13 @@ export function handleUpdatedCommitments(events: UpdatedCommitment[]): void {
 }
 
 export function handleDeletedCommitment(event: DeletedCommitment): void {
-  const commitment = loadCommitmentByMarketId(
-    event.params.lender,
-    event.params.marketId.toString(),
-    event.params.lendingToken
-  );
+  const commitmentId = event.params.commitmentId.toString();
+  const commitment = loadCommitment(commitmentId);
   commitment.committedAmount = BigInt.zero();
   commitment.expirationTimestamp = BigInt.zero();
   commitment.maxDuration = BigInt.zero();
   commitment.minAPY = BigInt.zero();
+  commitment.maxPrincipalPerCollateralAmount = BigInt.zero();
   commitment.save();
```

Comment:
lets try this instead so we can actually delete the entity from subgraph

```suggestion
 import { store } from '@graphprotocol/graph-ts'
...
  const commitmentId = event.params.commitmentId.toString();
  store.remove('Commitment', commitmentId);
```